### PR TITLE
1.0 Breaking Changes

### PR DESCRIPTION
**Pull request by dlockhart**
This is the much anticipated `1.0.0` release, where we can make some breaking changes.

Non-breaking changes:
- `node-config` and `browser-config` now use the `es2024` environment. This simply allows folks to use new ES2024 features without the linter complaining, but the features available [varies by Node version](https://compat-table.github.io/compat-table/es2016plus/) and obviously the browser vendor/version itself. Developers will continue to need to ensure that new browser features are properly transpiled by Babel.
- `common-config.js` has been merged into `index.js`. This shared "common" config was no longer shared by anything.

Breaking changes:
- ~`lit-config` now errors if language files have spaces around their keys or use inconsistent quotes for the keys. I know @bearfriend wants to apply [a few more things](https://github.com/Brightspace/awards-ui-lit/blob/729a6b85101e9ba9515d8bcd2bf69e6cf522947a/.eslintrc.json#L12-L20), but to keep this change manageable I left those out for now.~ Reverted these language changes, we'll look into them again later.
- `lit-config` has been updated to include: [lifecycle-super](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/lifecycle-super.md), [no-legacy-imports](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-legacy-imports.md), [no-native-attributes](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-native-attributes.md), [no-this-assign-in-render](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-this-assign-in-render.md) and [prefer-nothing](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/prefer-nothing.md). I didn't add [attribute-names](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/attribute-names.md) because it's impacting our internal reactive state properties, and also didn't add [prefer-static-styles](https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/prefer-static-styles.md) because it affects all `html` literals, not just elements -- which didn't seem right.
- All Polymer and Polymer-testing configs have been removed. Repositories using Polymer can remain on the `0.x` configuration until they upgrade to Lit.
- Renamed `open-wc-testing-config` to `testing-config` to better reflect its purpose.
- Removed `node: true` from `open-wc-testing-config` since these tests run in the browser, not in Node.
- `node-config` now has a `sourceType` of `module` by default, which means projects will be ESM by default. Any CommonJS projects can override this rule if desired.

### CODE REVIEW

File: lit-config.js [hunk 1]
```diff
@@ -46,23 +46,30 @@ module.exports = {
 	"overrides": [{
 		"files": "./**/lang/*.js",	
 		"rules": {
-			"quotes": 0
+			"quotes": "off",
+			"key-spacing": "error",
```
Comment (svanherk) [inline lit-config.js:50]: Minor: Do we want to use integers to match the rest of the file?

File: lit-config.js [hunk 2]
```diff
@@ -46,23 +46,30 @@ module.exports = {
...
+			"quote-props": ["error", "consistent"],
```
Comment (svanherk) [inline lit-config.js:51]: Seems like this rule is deprecated (although so is `quotes`). Are we worried about that?

Comment (svanherk) [reply lit-config.js:51]: Looks like we're supposed to migrate to [this plugin](https://eslint.style/packages/js) for some of our rules, which we'd probably want to do separate from this

**Review by svanherk — APPROVED** [review]

File: lit-config.js (hunk 1 above)
Comment (dlockhart) [reply lit-config.js:50]: I can yeah, although I find the integers difficult to remember which is which.

File: lit-config.js (hunk 2 above)
Comment (dlockhart) [reply lit-config.js:51]: Good catch. I'm just going to pull all these out and we can re-assess the language-specific stuff later.

File: lit-config.js (hunk 1 above)
Comment (svanherk) [reply lit-config.js]: Agreed, this is easier to read lol, I just like consistency a bit more. We could switch them all - we need a linter for our lint configs 😝

**Pull request merged**